crypto/bls/blst/bench_test.go

```diff
@@ -144,6 +146,37 @@ func BenchmarkVerify(b *testing.B) {
 	}
 }
 
+func BenchmarkParallelVerify(b *testing.B) {
```

Comment:
I think a setup work (fixed dataset) L155-L162 may be moved out to effectively measure the benchmark such as:

```go
func BenchmarkParallelVerify(b *testing.B) {
    L := benchAggregateLen
    tc := generateBenchmarkMaterialMulti(L)

    threads := runtime.NumCPU()
    jobs := make(chan int, len(tc.sigbs))
    for i := 0; i < L; i++ {
        jobs <- i
    }
    close(jobs) // for readability, channel close is directly placed here

    b.ResetTimer()
    for i := 0; i < b.N; i++ {
        var wg sync.WaitGroup
        wg.Add(threads)
        for i := 0; i < threads; i++ {
            go func() {
                for i := range jobs {
                    sig, _ := SignatureFromBytes(tc.sigbs[i])
                    pk, _ := PublicKeyFromBytes(tc.pkbs[i])
                    Verify(sig, tc.msgs[i][:], pk)
                }
                wg.Done()
            }()
        }
        wg.Wait()
    }
}
```

The operation cost seems reduced to 2/3.

Reply:
I thought the `jobs` queue must be refilled every iteration, isn't it?

Reply:
Aha, exactly. Sorry for the noise. LGTM.